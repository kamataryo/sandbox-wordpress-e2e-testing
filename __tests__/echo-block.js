const {
  createNewPost,
  getEditedPostContent
} = require('@wordpress/e2e-test-utils')

jest.setTimeout(60000) // タイムアウトを長めに設定

// insertBlocks は e2e-test-utils に含まれているが、動かないので自分でユーティリティを作る
const insertBlock = async searchTerm => {
  await page.click('.edit-post-header [aria-label="Add block"]')
  await page.waitForSelector('.editor-inserter__menu')
  await page.keyboard.type(searchTerm)
  await page.click(`button[aria-label="${searchTerm}"]`)
}

describe('Echo ブロックのテスト', () => {
  beforeEach(async () => await createNewPost())

  test('エコーブロックで作成される Post の内容がスナップショットと一致する', async () => {
    await insertBlock('Echo Block')
    await page.keyboard.type('Hello')
    await page.keyboard.press('Enter')
    expect(await getEditedPostContent()).toMatchSnapshot()
  })
})
