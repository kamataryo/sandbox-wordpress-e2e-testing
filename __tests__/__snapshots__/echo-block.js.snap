// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Echo ブロックのテスト エコーブロックで作成される Post の内容がスナップショットと一致する 1`] = `
"<!-- wp:register-echo-block/echo-block -->
<div class=\\"wp-block-register-echo-block-echo-block\\"><span style=\\"font-size:1em\\" class=\\"value\\">Hello</span><span style=\\"font-size:.75em\\">Hello</span><span style=\\"font-size:.5em\\">Hello</span><span style=\\"font-size:.25em\\">Hello</span>...</div>
<!-- /wp:register-echo-block/echo-block -->"
`;
